tools/build_defs/pkg/pkg.bzl

```diff
@@ -49,14 +48,18 @@ def _pkg_tar_impl(ctx):
         "--owner_name=" + ctx.attr.ownername,
     ]
 
-    file_inputs = ctx.files.srcs[:]
-
     # Add runfiles if requested
     if ctx.attr.include_runfiles:
+        runfiles_depsets = []
         for f in ctx.attr.srcs:
-            if hasattr(f, "default_runfiles"):
-                run_files = f.default_runfiles.files.to_list()
-                file_inputs += run_files
+            default_runfiles = f[DefaultInfo].default_runfiles
```

Comment:
This accesses the same information via a provider rather than via attribute (deprecated way to do that?).
Documentation says `default_runfiles` may return `None` so I've added a check.
https://docs.bazel.build/versions/master/skylark/lib/DefaultInfo.html#default_runfiles